### ISSUE TITLE
Trying to use tf.layers.embedding as starting layer results in expected embedding_Embedding1 to have 3 dimension(s)

### ISSUE THREAD

**Issue by Lokesh-K-Haralakatta** [tensorflow/tfjs#783]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### TensorFlow.js version
https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@0.13.0
#### Browser version
N/A
#### Describe the problem
Trying to use tf.layers.embedding as a starting layer for a simple example but always results with the error while trying to train the model as pasted below:

Uncaught (in promise) Error: Error when checking target: expected embedding_Embedding1 to have 3 dimension(s). but got array with shape 6,1
    at new t (tfjs@0.13.0:2)
    at standardizeInputData (tfjs@0.13.0:2)
    at t.standardizeUserData (tfjs@0.13.0:2)
    at t.<anonymous> (tfjs@0.13.0:2)
    at tfjs@0.13.0:2
    at Object.next (tfjs@0.13.0:2)
    at tfjs@0.13.0:2
    at new Promise (<anonymous>)
    at __awaiter$1 (tfjs@0.13.0:2)
    at t.fit (tfjs@0.13.0:2)

#### Code to reproduce the bug:

//Sample data to train
const xs = tf.tensor2d([[1],[2],[3],[4],[5],[6]]);
const ys = tf.tensor2d([[1],[0],[1],[0],[1],[0]]);

// Define a model
const model = tf.sequential();

//Add layers to model
model.add(tf.layers.embedding({
                                    inputDim: xs.shape[0],
                                    outputDim: xs.shape[1],
                                    trainable: false
                                  }));

model.add(tf.layers.dense({units: 1, activation: 'sigmoid'}));

// Train the model using the data
await model.fit(xs, ys, { batchSize: 2,epochs: 20 });

Please have a look and let me know anything is missed in the above given code sample due to which the error is being thrown.

**Comment by bileschi** [comment]
The API for tf.layers.embedding requires inputs be rank-2 (rank-3 including the batch).  What happened here is the layer received an input (xs) which is rank 2.

inputDIm should be the size of the vocabulary
outputDim should be the dimensionality of the vectorspace produced.

If you consider the inputs as sequences of length 1, this code should work: (note the additional square brackets)

//Sample data to train
const xs = tf.tensor3d([[[1]],[[2]],[[3]],[[4]],[[5]],[[6]]]);
const ys = tf.tensor2d([[1],[0],[1],[0],[1],[0]]);

// Define a model
const model = tf.sequential();

//Add layers to model
model.add(tf.layers.embedding({
inputDim: 6,
outputDim: 1,
trainable: true
}));
model.add(tf.layers.dense({units: 1, activation: 'sigmoid'}));

// print out model summary
model.summary():

// compile
model.compile({loss: "meanSquaredError", optimizer: "sgd"});
// fit

model.fit(xs, ys);
```________________________________________________________________
tfjs@0.13.0:2 Layer (type)                 Output shape              Param #   
tfjs@0.13.0:2 =================================================================
tfjs@0.13.0:2 embedding_Embedding1 (Embedd [null,null,1]             6         
tfjs@0.13.0:2 _________________________________________________________________
tfjs@0.13.0:2 dense_Dense1 (Dense)         [null,null,1]             2         
tfjs@0.13.0:2 =================================================================
tfjs@0.13.0:2 Total params: 8
tfjs@0.13.0:2 Trainable params: 2
tfjs@0.13.0:2 Non-trainable params: 6
tfjs@0.13.0:2 
```
